components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java

```diff
@@ -870,4 +871,27 @@ public static void updateEveryOneRoleV2MetaData(int tenantId) {
             }
         }
     }
+
+    /**
+     * Update system role meta data.
+     *
+     * @param tenantId Tenant Id.
+     */
+    public static void updateSystemRoleV2MetaData(int tenantId) {
+
+        // Handle system role creation also here if legacy runtime is disabled.
+        if (!CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME) {
+            try {
+                UserStoreManager userStoreManager = (UserStoreManager) SCIMCommonComponentHolder.getRealmService().
+                        getTenantUserRealm(tenantId).getUserStoreManager();
+                String domainName = UserCoreUtil.getDomainName(userStoreManager.getRealmConfiguration());
+                SCIMGroupHandler scimGroupHandler = new SCIMGroupHandler(userStoreManager.getTenantId());
+                String systemRoleNameWithDomain =
+                        UserCoreUtil.addDomainToName(AccountConstants.ACCOUNT_LOCK_BYPASS_ROLE, domainName);
```

Comment:
Seems domain addition is unnecessary

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/500/commits/34bf52581a534f4db70e211520e69d6e13a58550